pkg/client/cli/telepresence_test.go

```diff
@@ -413,26 +413,49 @@ func (ts *telepresenceSuite) TestB_Connected() {
 }
 
 func (ts *telepresenceSuite) TestC_Uninstall() {
+	telepresence(ts.T(), "connect")
+
 	ts.Run("Uninstalls the traffic manager and quits", func() {
 		require := ts.Require()
 		ctx := dlog.NewTestContext(ts.T(), false)
+
 		names := func() (string, error) {
 			return ts.kubectlOut(ctx, "get",
 				"--namespace", ts.managerTestNamespace,
 				"svc,deploy", "traffic-manager",
 				"--ignore-not-found",
 				"-o", "jsonpath={.items[*].metadata.name}")
 		}
+
 		stdout, err := names()
 		require.NoError(err)
 		require.Equal(2, len(strings.Split(stdout, " "))) // The service and the deployment
 		ts.NoError(ts.capturePodLogs(ctx, "traffic-manager", ts.managerTestNamespace))
 
+		// Add webhook agent to test webhook uninstall
+		jobname := "echo-auto-inject"
+		deployname := "deploy/" + jobname
+		require.NoError(ts.applyApp(ctx, jobname, jobname, 80))
+		require.NoError(ts.kubectl(ctx, "rollout", "status", "-w", deployname, "-n", ts.namespace))
+		stdout, stderr := telepresence(ts.T(), "list", "--namespace", ts.namespace, "--agents")
+		require.Empty(stderr)
+		require.Contains(stdout, jobname+": ready to intercept (traffic-agent already installed)")
+
 		// The telepresence-test-developer will not be able to uninstall everything
 		require.NoError(run(ctx, "kubectl", "config", "use-context", "default"))
-		stdout, stderr := telepresence(ts.T(), "uninstall", "--everything")
+		stdout, stderr = telepresence(ts.T(), "uninstall", "--everything")
 		require.Empty(stderr)
 		require.Contains(stdout, "Root Daemon quitting... done")
+
+		// Double check webhook agent is uninstalled
+		require.NoError(ts.kubectl(ctx, "rollout", "status", "-w", deployname, "-n", ts.namespace))
+		stdout, stderr = telepresence(ts.T(), "list", "--namespace", ts.namespace)
+		require.Empty(stderr)
+		require.Contains(stdout, jobname+": ready to intercept (traffic-agent not yet installed)")
+		stdout, stderr = telepresence(ts.T(), "uninstall", "--everything")
+		require.Empty(stderr)
+		require.Contains(stdout, "Root Daemon quitting... done")
```

Comment:
why re-run the uninstall? does `telepresence list` re-install stuff? If so, could we `kubectl get` the pod instead and verify that it doesn't have the sidecar?

Reply:
> does telepresence list re-install stuff?


yes

> If so, could we kubectl get the pod instead?

We could, i think it would require a regex to find `echo-auto-inject-<SHA>             1/1`. Not a difficult regex, I could make this change